File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 20 and 21 April 2019/05. Easter Eggs/05. Easter Eggs.js
function easterEggs(arg) {

    let loop = +arg.shift()
    let eggsDict = {'red': 0, 'orange': 0, 'blue': 0, 'green': 0}

    for (let i = 0; i < loop; i++) {
        color = arg.shift()
        eggsDict[color] += 1
    }

    console.log(`Red eggs: ${eggsDict["red"]}`)
    console.log(`Orange eggs: ${eggsDict["orange"]}`)
    console.log(`Blue eggs: ${eggsDict["blue"]}`)
    console.log(`Green eggs: ${eggsDict["green"]}`)

    const sortEggsDict = Object.entries(eggsDict).sort((a, b) => b[1] - a[1])
    console.log(`Max eggs: ${sortEggsDict[0][1]} -> ${sortEggsDict[0][0]}`)

}

//easterEggs([7, 'orange', 'blue', 'green', 'green', 'blue', 'red', 'green'])

File: Python Exams/01. Basic Exam Collection/Basics Online Exam - 20 and 21 April 2019/05. Easter Eggs/05. Easter Eggs.py
loop = int(input())
eggs_dict = {'red': 0, 'orange': 0, 'blue': 0, 'green': 0}

for _ in range(loop):
    color = input()
    eggs_dict[color] += 1

print(f'Red eggs: {eggs_dict["red"]}')
print(f'Orange eggs: {eggs_dict["orange"]}')
print(f'Blue eggs: {eggs_dict["blue"]}')
print(f'Green eggs: {eggs_dict["green"]}')

eggs_dict = sorted(eggs_dict.items(), key= lambda x: -x[1])[0]
print(f'Max eggs: {eggs_dict[1]} -> {eggs_dict[0]}')



'''
7
orange
blue
green
green
blue
red
green
'''
